### PR TITLE
chore(ci): developing ci with docker build tool 

### PR DESCRIPTION
**Pull request by everpcpc**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

build debug with build tool docker image on ci
separate build for different workflow dependency

## Changelog

- Build/Testing/CI

## Related Issues

need: #4604
for: #3611 #3844 
fix: #4595

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
@mergify update

File: .github/workflows/developing.yml [hunk 1]
```diff
@@ -29,89 +29,90 @@ jobs:
...
-          - { os: ubuntu-latest, target: x86_64-unknown-linux-gnu }
+        target:
+          - x86_64-unknown-linux-gnu
+          - aarch64-unknown-linux-gnu
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/build_debug
+        with:
+          target: ${{ matrix.target }}
+
```
Comment (Xuanwo) [inline .github/workflows/developing.yml:60]:
```suggestion
```

**Review by Xuanwo — COMMENTED** [review]
Mostly LGTM

**Review by ZhiHanZ — APPROVED** [review]
LGTM, if build patch become bigger, we should introduce some self hosted runner @everpcpc

**Pull request merged**